mono/eglib/gspawn.c

```diff
@@ -72,7 +72,13 @@
 #define set_error_cond(cond,msg, ...) do { if ((cond) && gerror != NULL) *gerror = g_error_new (G_LOG_DOMAIN, 1, msg, __VA_ARGS__); } while (0)
 #define set_error_status(status,msg, ...) do { if (gerror != NULL) *gerror = g_error_new (G_LOG_DOMAIN, status, msg, __VA_ARGS__); } while (0)
 #define NO_INTR(var,cmd) do { (var) = (cmd); } while ((var) == -1 && errno == EINTR)
-#define CLOSE_PIPE(p) do { close (p [0]); close (p [1]); } while (0)
+
+static void
```

Comment:
Mark as inline? Does it get inlined anyway?

Reply:
This code is creating proceses, for purposes of debugging, and reading pipes.
So it doesn't matter.
Inlining would be a size deoptimization, therefore maybe slower. Let the compiler decide.